dev/Interop/StoragePickers/PickerCommon.cpp

```diff
@@ -0,0 +1,213 @@
+            knownFolderId = FOLDERID_MusicLibrary;
+            break;
+        case winrt::Microsoft::Windows::Storage::Pickers::PickerLocationId::PicturesLibrary:
+            knownFolderId = FOLDERID_PicturesLibrary;
+            break;
+        case winrt::Microsoft::Windows::Storage::Pickers::PickerLocationId::VideosLibrary:
+            knownFolderId = FOLDERID_VideosLibrary;
+            break;
+        case winrt::Microsoft::Windows::Storage::Pickers::PickerLocationId::Objects3D:
+            knownFolderId = FOLDERID_Objects3D;
+            break;
+        default:
+            return nullptr;
+        }
+
+        auto knownFolderManager = winrt::create_instance<IKnownFolderManager>(CLSID_KnownFolderManager);
+
+        winrt::com_ptr<IKnownFolder> knownFolder{};
+        winrt::hresult hr = knownFolderManager->GetFolder(knownFolderId, knownFolder.put());
+        if (!knownFolder)
+        {
+            knownFolderManager->GetFolder(FOLDERID_Documents, knownFolder.put());
+        }
+        if (!knownFolder)
+        {
+            return nullptr;
+        }
+
+        winrt::com_ptr<IShellItem> defaultFolder{};
+        hr = knownFolder->GetShellItem(0, IID_PPV_ARGS(defaultFolder.put()));
+        if (FAILED(hr))
+        {
+            return nullptr;
+        }
+
+        return defaultFolder;
+    }
+
+
+    winrt::hstring FormatExtensionWithWildcard(winrt::hstring extension)
+    {
+        if (!extension.empty() && extension[0] == L'*')
+        {
+            return extension;
+        }
+        else
+        {
+            return L"*" + extension;
+        }
+    }
+
+    winrt::hstring JoinExtensions(winrt::Windows::Foundation::Collections::IVectorView<winrt::hstring> extensions)
+    {
+        winrt::hstring result;
+        bool first = true;
+        for (const auto& ext : extensions)
+        {
+            if (first)
+            {
+                result = FormatExtensionWithWildcard(ext);
+                first = false;
+            }
+            else
+            {
+                result = result + L";" + FormatExtensionWithWildcard(ext);
+            }
+        }
+        return result;
+    }
+}
+
+
+namespace PickerCommon {
+
+    using namespace winrt;
+
+    bool IsHStringNullOrEmpty(winrt::hstring value)
+    {
+        return value.empty();
+    }
+
+    winrt::hstring GetPathFromShellItem(winrt::com_ptr<IShellItem> shellItem)
+    {
+        wil::unique_cotaskmem_string filePath;
+        check_hresult(shellItem->GetDisplayName(SIGDN_FILESYSPATH, filePath.put()));
+        return winrt::hstring{ filePath.get() };
+    }
+
+    std::vector<COMDLG_FILTERSPEC> CaptureFilterSpec(std::vector<winrt::hstring>& buffer, winrt::Windows::Foundation::Collections::IVectorView<winrt::hstring> filters)
+    {
+        std::vector<COMDLG_FILTERSPEC> result(filters.Size());
+        buffer.clear();
+        buffer.reserve(filters.Size() * (size_t)2);
+        for (auto filter : filters)
```

Comment:
>auto [](http://example.com/codeflow?start=13&length=4)

nit: `const auto&` like the other CaptureFilterSpec?

Reply:
Thank you for your suggestion, fixed to use const reference